### DISCUSSION TITLE
How to configure the PHP command line on Windows?

### DISCUSSION THREAD

**Discussion by nzyeehaa** [datenstrom/community#683] [Ask a question] [answered]
I have never used the [command](https://github.com/datenstrom/yellow-extensions/tree/master/source/command) or [serve](https://github.com/datenstrom/yellow-extensions/tree/master/source/serve) extensions - (which are installed by default when you install yellow) - as I run a development server on windows, and normally delete _command / serve / meta & bundle_ from my yellow installs, but today I decided to have a play.

...and nothing worked...

first - you need php on your machine as mentioned in the [command extension](https://github.com/datenstrom/yellow-extensions/tree/master/source/command) (which i have)
second - _serve_ needs _command_ installed or _it_ wont work (both installed by default)
third - on a windows machine - php doesn't natively work at the command line until you **add it to your path** (and restart any terminal windows)

basic steps for windows 10...
control panel...
search "env"
edit environment variables
select _path_ and _->edit_
_->new_ and add the path to the folder where php.exe resides ( e.g: Y:\Server\php-7.3 )
restart any terminal windows, navigate to where your yellow install is located and carry on as per _command / serve_ instructions

hope it helps someone.

**Comment by schulle4u** [comment]
As Microsoft [has dropped php support from version 8.0](https://news-web.php.net/php.internals/110907), theres also the WSL (Windows subsystem for linux) as an alternative development setup. Just activate it using the old control panel (programs and features -> activate/deactivate windows features),  and download any of the distributions provided in the microsoft store. Local drives are mapped automatically, e. g. drive C is located at `/mnt/c/`, so you should be able to use your desired working directory.

**Reply by nzyeehaa** [reply]
wow - thanks for this... 
I was always going to setup dual boot with linux mint, but couldn't be bothered with MicSoft menstrual tantrums, so it never got done.

WSL 2 looks awesome at up to 20 times faster than v1...
very handy.

**Reply by IngoBelka** [reply]
For me a Vbox-Linux-machine is the best way, to host my developement environment, even because the most web servers are Linux machines.

**Comment by annaesvensson** [comment, marked as answer]
Thank you. 👍 Marking as answered/solved.
